routing.go

```diff
@@ -629,13 +615,13 @@ func (dht *IpfsDHT) findProvidersAsyncRoutine(ctx context.Context, key multihash
 
 			return peers, nil
 		},
-		func(peerset *kpeerset.SortedPeerset) bool {
+		func() bool {
 			return ps.Size() >= count
 		},
 	)
 
 	if err != nil && ctx.Err() == nil {
```

Comment:
I suspect we could bundle this check into the `lookup.completed` variable, but haven't gotten around to it yet.